providers/common/digest_to_nid.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/objects.h>
+#include <openssl/core_names.h>
+#include <openssl/evp.h>
+#include <openssl/core.h>
+#include "prov/check.h"
+#include "internal/nelem.h"
+
+/*
+ * Internal library code deals with NIDs, so we need to translate from a name.
+ * We do so using EVP_MD_is_a(), and therefore need a name to NID map.
+ */
+int digest_md_to_nid(const EVP_MD *md, const OSSL_ITEM *it, size_t it_len)
+{
+    size_t i;
+    int mdnid = NID_undef;
+
+    if (md == NULL)
```

Comment:
Not a fan of run-time NUL checks for args.  But seems simplest to say 'return NID_undef' here, avoid the goto and the label.

Reply:
Got rid of the end.
I disagree with your thoughts on NULL checks.